lib/chef/provider/package/yum.rb

```diff
@@ -1220,7 +1220,18 @@ def parse_arch
         # There is currently no support for filename matching.
         def parse_dependency
           # Transform the package_name into a requirement
-          yum_require = RPMRequire.parse(@new_resource.package_name)
+
+          # If we are passed a version we have to assume it's a requirement first. If it can't be
+          # parsed only yum_require.name will be set and @new_resource.version will be left intact
+          if @new_resource.version
```

Comment:
add a comment that version is/may be a version constraint instead of just a version